be/src/vec/aggregate_functions/aggregate_function_java_udaf.h

```diff
@@ -0,0 +1,361 @@
+                                            s.get_error_msg()));
+            }
+            return s;
+        };
+
+        RETURN_IF_ERROR(register_id("<init>", UDAF_EXECUTOR_CTOR_SIGNATURE, executor_ctor_id));
+        RETURN_IF_ERROR(register_id("add", UDAF_EXECUTOR_ADD_SIGNATURE, executor_add_id));
+        RETURN_IF_ERROR(register_id("close", UDAF_EXECUTOR_CLOSE_SIGNATURE, executor_close_id));
+        RETURN_IF_ERROR(register_id("merge", UDAF_EXECUTOR_MERGE_SIGNATURE, executor_merge_id));
+        RETURN_IF_ERROR(
+                register_id("serialize", UDAF_EXECUTOR_SERIALIZE_SIGNATURE, executor_serialize_id));
+        RETURN_IF_ERROR(
+                register_id("getValue", UDAF_EXECUTOR_RESULT_SIGNATURE, executor_result_id));
+        serialize_len_id = env->GetFieldID(executor_cl, "serializeLength", "J");
+        RETURN_ERROR_IF_EXC(env);
+        return Status::OK();
+    }
+
+public:
+    jclass executor_cl;
+    jobject executor_obj;
+    jmethodID executor_ctor_id;
+
+    jmethodID executor_add_id;
+    jmethodID executor_merge_id;
+    jmethodID executor_serialize_id;
+    jmethodID executor_result_id;
+    jmethodID executor_close_id;
+    jfieldID serialize_len_id;
+
+    std::unique_ptr<int64_t[]> input_values_buffer_ptr;
+    std::unique_ptr<int64_t[]> input_nulls_buffer_ptr;
+    std::unique_ptr<int64_t[]> input_offsets_ptrs;
+    std::unique_ptr<int64_t> output_value_buffer;
+    std::unique_ptr<int64_t> output_null_value;
+    std::unique_ptr<int64_t> output_offsets_ptr;
+    std::unique_ptr<int64_t> output_intermediate_state_ptr;
+
+    bool first_init;
+    int argument_size = 0;
+    std::string serialize_data;
+};
+
+class AggregateJavaUdaf final
+        : public IAggregateFunctionDataHelper<AggregateJavaUdafData, AggregateJavaUdaf> {
+public:
+    AggregateJavaUdaf(const TFunction& fn, const DataTypes& argument_types, const Array& parameters,
+                      const DataTypePtr& return_type)
+            : IAggregateFunctionDataHelper(argument_types, parameters),
+              fn_(fn),
+              return_type_(return_type) {}
+    ~AggregateJavaUdaf() = default;
+
+    static AggregateFunctionPtr create(const TFunction& fn, const DataTypes& argument_types,
+                                       const Array& parameters, const DataTypePtr& return_type) {
+        return std::make_shared<AggregateJavaUdaf>(fn, argument_types, parameters, return_type);
+    }
+
+    void create(AggregateDataPtr __restrict place) const override {
+        new (place) Data(argument_types.size());
+    }
+
+    String get_name() const override { return fn_.name.function_name; }
+
+    DataTypePtr get_return_type() const override { return return_type_; }
+
+    void add(AggregateDataPtr __restrict place, const IColumn** columns, size_t row_num,
+             Arena*) const override {
+        Status status = Status::OK();
+        RETURN_IF_STATUS_ERROR(status, data(place).init_udaf(fn_));
```

Comment:
why not do init in `create` function and delete the `first_init`?

Reply:
Have test in create function do init operator, but it's unable to get jvm